src/server/mod.rs

```diff
@@ -315,6 +329,20 @@ impl<State: Send + Sync + 'static> Server<State> {
         res.map_err(|e| std::io::Error::new(std::io::ErrorKind::Other, e))?;
         Ok(())
     }
+
+    /// Convert the server into a mock server.
+    ///
+    /// Unlike `into_http_service` this method enables default middleware.
+    ///
+    /// # Stability
+    ///
+    /// This method will be expanded in the future to allow mocking requests
+    /// in a more direct manner.
+    pub fn into_mock(mut self) -> io::Result<http_service_mock::TestBackend<Service<State>>> {
+        self.enable_default_middleware();
+        let server = http_service_mock::make_server(self.into_http_service())?;
+        Ok(server)
+    }
```

Comment:
By providing separate method for nesting, we don't need to provide a "special" wrapper method like this.